eclair-core/src/test/scala/fr/acinq/eclair/blockchain/bitcoind/BitcoinCoreClientSpec.scala

```diff
@@ -978,4 +978,59 @@ class BitcoinCoreClientSpec extends TestKitBaseClass with BitcoindService with A
+    val walletPubKey = sender.expectMsgType[PublicKey]
+    val miner = new BitcoinCoreClient(bitcoinrpcclient)
+    miner.getP2wpkhPubkey().pipeTo(sender.ref)
+    val nonWalletPubKey = sender.expectMsgType[PublicKey]
+    // We use a large input script to be able to fill the mempool with a few transactions.
+    val bigInputScript = Script.write(Seq.fill(200)(Seq(OP_PUSHDATA(ByteVector.fill(15)(42)), OP_DROP)).flatten)
+    val largeInputsCount = 110
+    // We prepare confirmed parent transactions containing such inputs.
+    val parentTxs = (walletPubKey +: Seq.fill(12)(nonWalletPubKey)).map(recipient => {
+      val mainOutput = TxOut(500_000 sat, Script.pay2wpkh(recipient))
+      val outputsWithLargeScript = Seq.fill(largeInputsCount)(TxOut(1_000 sat, Script.pay2wsh(bigInputScript)))
+      val outputs = mainOutput +: outputsWithLargeScript
+      val txNotFunded = Transaction(2, Nil, mainOutput +: outputsWithLargeScript, 0)
+      miner.fundTransaction(txNotFunded, FundTransactionOptions(FeeratePerKw(500 sat), changePosition = Some(outputs.length))).pipeTo(sender.ref)
+      val fundedTx = sender.expectMsgType[FundTransactionResponse].tx
+      miner.signTransaction(fundedTx, allowIncomplete = false).pipeTo(sender.ref)
+      val signedTx = sender.expectMsgType[SignTransactionResponse].tx
+      miner.publishTransaction(signedTx).pipeTo(sender.ref)
+      sender.expectMsg(signedTx.txid)
+      signedTx
+    })
+    generateBlocks(1)
+
+    def publishLargeTx(parentTx: Transaction, amount: Satoshi, wallet: BitcoinCoreClient): Transaction = {
+      val mainInput = TxIn(OutPoint(parentTx, 0), Nil, 0)
+      val inputsWithLargeScript = (1 to largeInputsCount).map(i => TxIn(OutPoint(parentTx, i), ByteVector.empty, 0, ScriptWitness(Seq(ByteVector(1), bigInputScript))))
+      val txIn = mainInput +: inputsWithLargeScript
+      val txOut = Seq(TxOut(amount, Script.pay2wpkh(randomKey().publicKey)))
+      wallet.signTransaction(Transaction(2, txIn, txOut, 0), allowIncomplete = true).pipeTo(sender.ref)
+      val signedTx = sender.expectMsgType[SignTransactionResponse].tx
+      assert(390_000 <= signedTx.weight() && signedTx.weight() <= 400_000) // standard transactions cannot exceed 400 000 WU
+      wallet.publishTransaction(signedTx).pipeTo(sender.ref)
+      sender.expectMsg(signedTx.txid)
+      signedTx
+    }
+
+    // We create a large unconfirmed transaction with a low feerate.
+    val tx = publishLargeTx(parentTxs.head, 450_000 sat, wallet)
+    // Transactions with higher feerates are added to the mempool and evict the first transaction.
+    parentTxs.tail.foreach(parentTx => publishLargeTx(parentTx, 300_000 sat, miner))
+    // Even though the wallet transaction has been evicted, bitcoind doesn't double-spend its inputs.
+    wallet.getMempoolTx(tx.txid).pipeTo(sender.ref)
+    assert(sender.expectMsgType[Failure].cause.getMessage.contains("Transaction not in mempool"))
+    wallet.listUnspent().pipeTo(sender.ref)
```

Comment:
nit: you could use the low-level `gettransaction`  RPC call to check that tx is till in the wallet ?

Reply:
Done in https://github.com/ACINQ/eclair/pull/2529/commits/89dc986b66b7629bf542b45cf1a98441108e755f

Reply:
Actually this is using `getrawtransaction` and seems to be flaky, I'll try `gettransaction` and see if it's better.

Reply:
Yes I think `gettransaction` is not exposed in our RPC client yet, you either need to add this method or call `invoke` directly and parse the returned JSON object

Reply:
Done in https://github.com/ACINQ/eclair/pull/2529/commits/373171a2131f224d8658079f671ab49dfb286092